Library/Homebrew/language/python.rb

```diff
@@ -139,11 +139,45 @@ def virtualenv_create(venv_root, python = "python", formula = self)
         venv
       end
 
+      def needs_python2?
+        requirements.any? do |req|
+          req.is_a?(PythonRequirement) && !req.is_a?(Python3Requirement) &&
+            (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == "python" && (dep.required? || build.with?(dep))
+          end
+      end
+
+      def needs_python3?
+        requirements.any? do |req|
+          req.is_a?(Python3Requirement) && (req.required? || build.with?(req))
+        end ||
+          deps.any? do |dep|
+            dep.name == "python3" && (dep.required? || build.with?(dep))
+          end
+      end
+
       # Helper method for the common case of installing a Python application.
       # Creates a virtualenv in `libexec`, installs all `resource`s defined
       # on the formula, and then installs the formula.
-      def virtualenv_install_with_resources
-        venv = virtualenv_create(libexec)
+      def virtualenv_install_with_resources(*args)
+        options = args[-1].is_a?(Hash) ? args.pop : {}
```

Comment:
We tend to prefer `args.last` instead of `args[-1]`. However, I wonder why we need this handling at all if an `options = {}` in the method signature will do just fine. It's not like we need the flexibility of mixing an arbitrary number of non-option arguments (currently none) and option arguments right now.